test/jasmine/assets/transforms/filter.js

```diff
@@ -0,0 +1,161 @@
+    filtersrc: {
+        valType: 'enumerated',
+        values: ['x', 'y'],
+        dflt: 'x'
+    }
+};
+
+/**
+ * Supply transform attributes defaults
+ *
+ * @param {object} transformIn
+ *  object linked to trace.transforms[i] with 'type' set to exports.name
+ * @param {object} fullData
+ *  the plot's full data
+ * @param {object} layout
+ *  the plot's (not-so-full) layout
+ *
+ * @return {object} transformOut
+ *  copy of transformIn that contains attribute defaults
+ */
+exports.supplyDefaults = function(transformIn, fullData, layout) {
+    var transformOut = {};
+
+    function coerce(attr, dflt) {
+        return Lib.coerce(transformIn, transformOut, exports.attributes, attr, dflt);
+    }
+
+    coerce('operation');
+    coerce('value');
+    coerce('filtersrc');
+
+    // or some more complex logic using fullData and layout
+
+    return transformOut;
+};
+
+/**
+ * Apply transform !!!
+ *
+ * @param {array} data
+ *  array of transformed traces (is [fullTrace] upon first transform)
+ *
+ * @param {object} state
+ *  state object which includes:
+ *      - transform {object} full transform attributes
+ *      - fullTrace {object} full trace object which is being transformed
+ *      - fullData {array} full pre-transform(s) data array
+ *      - layout {object} the plot's (not-so-full) layout
+ *
+ * @return {object} newData
+ *  array of transformed traces
+ */
+exports.transform = function(data, state) {
+
+    // one-to-one case
+
+    var newData = data.map(function(trace) {
+        return transformOne(trace, state);
+    });
+
+    return newData;
+};
+
+function transformOne(trace, state) {
+    var newTrace = Lib.extendDeep({}, trace);
+
+    var opts = state.transform;
+    var src = opts.filtersrc;
+    var filterFunc = getFilterFunc(opts);
+    var len = trace[src].length;
+    var arrayAttrs = findArrayAttributes(trace);
+
+    arrayAttrs.forEach(function(attr) {
+        Lib.nestedProperty(newTrace, attr).set([]);
+    });
+
+    function fill(attr, i) {
+        var arr = Lib.nestedProperty(trace, attr).get();
+        var newArr = Lib.nestedProperty(newTrace, attr).get();
+
+        newArr.push(arr[i]);
+    }
+
+    for(var i = 0; i < len; i++) {
+        var v = trace[src][i];
+
+        if(!filterFunc(v)) continue;
+
+        for(var j = 0; j < arrayAttrs.length; j++) {
+            fill(arrayAttrs[j], i);
+        }
+    }
+
+    return newTrace;
+}
+
+function getFilterFunc(opts) {
+    var value = opts.value;
+
+    switch(opts.operation) {
+        case '=':
+            return function(v) { return v === value; };
+        case '<':
+            return function(v) { return v < value; };
+        case '>':
+            return function(v) { return v > value; };
+    }
+}
+
+function findArrayAttributes(obj, root) {
+    root = root || '';
+
+    var list = [];
+
+    Object.keys(obj).forEach(function(k) {
+        var val = obj[k];
+
+        if(k.charAt(0) === '_') return;
+
+        if(k === 'transforms') {
```

Comment:
I think `k === 'colorscales'` might need a special case too. I'm not using this plugin but adapted this approach for my own purposes, and I found that when coloring by a continuous variable you end up with errors unless colorscales are left alone. (And just by inspecting the docs, perhaps `ticktext` and `tickvals` might be special as well.)